### PR TITLE
fix security

### LINKED ISSUE #18: Insecure?
When installed this allowes all users to send mails to anybody. This is not secure and can be abused for spam or mails that are send in your or your company names!

I can provide a full working example if needed.

### PR DESCRIPTION
**Pull request by mathe42**
fixes #18 
replacement for #20 as it doesn't fix issue.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,15 +4,27 @@ import nodemailer from 'nodemailer'
 
 export default function (moduleOptions) {
   const options = { ...this.options.mail, ...moduleOptions }
+
   if (!options.smtp) {
     throw new Error('SMTP config is missing.')
   }
+
+  if(!options.mail || (!options.mail.to && !options.mail.cc && !options.mail.bcc)) {
```
Comment (mathe42) [inline src/index.js:12]: Just some validation

File: src/index.js [hunk 2]
```diff
@@ -4,15 +4,27 @@ import nodemailer from 'nodemailer'
...
+    throw new Error('You have to provide to/cc/bcc in config.')
+  }
+
   const app = express()
   const transport = nodemailer.createTransport(options.smtp)
   app.use(express.json())
   app.post('/send', async (req, res) => {
     try {
-      await transport.sendMail(req.body)
+      await transport.sendMail({
```
Comment (mathe42) [inline src/index.js:21]: use req.body but overwirte to, cc and bcc with empty (so no injection from client posible) and overwrite it with default values.

**Review by mathe42 — COMMENTED** [review]
With this config has to have a mail object where ALL options from client can be overwriten.

File: src/index.js (hunk 2 above)
Comment (dword-design) [reply src/index.js:19]: Nice. I've simplified this a bit with optional chaining.

**Pull request merged**